### ISSUE TITLE
Upvote/downvote promotions

### ISSUE THREAD

**Issue by ericrkuo** [ubclaunchpad/PromoPal-backend#104]
Work that needs to be done
- need the design team to come up with a design for this
- when we call GET promotions or GET promotions/:id we want to know which user is requesting this
  - this issue is blocked by https://github.com/ubclaunchpad/foodies/issues/69
  - this is so that we can send information back whether the user upvoted/downvoted the promotion already
  - can be accomplished by creating a join table (many to many relationship between user and promotion) that indicates whether a user already upvoted or downvoted a promotion
E.g. BE work
``` 
userId | promotionId | vote (vote can be an enum)
id1      | id2                |   1       // means upvoted 
id1      | id3                |   -1     // means downvoted
// non-existent tuple means did not upvote or downvote
```

Base work already done
- https://github.com/ubclaunchpad/foodies/issues/49
  - Added column to promotion to keep track of votes and added the ability to upvote/downvote a promotion

**Comment by ld357** [comment]
Just an idea to support but maybe we can have some functionality that states a user "tried" this promotion, and once they click that button or something, we'll ask them to upvote or downvote?

Also I'm thinking if we should just do upvote itself since the idea is to support local- we don't want to create a negative reputation for any restaurant's promotion on our app

**Comment by ld357** [comment]
Also expanding on that, if we find ourselves in a scenario where we only have promotions with a bunch of downvotes 😅  seems kinda risky
